sdk/attestation/attestation/src/attestationClient.ts

```diff
@@ -197,7 +197,14 @@ export class AttestationClient {
       } else {
         options = credentialsOrOptions;
       }
-    } else if (clientOptions !== undefined) {
+    }
+
+    // If arg3 is defined, it has to be clientOptions and arg2 has to be a token credential
```

Comment:
I think we can also ditch the initial assignment to `options`, since `clientOptions` has a default value that isn't `undefined`

I think we can simplify everything in this constructor to something like

```ts
if(credentialsOrOptions && isTokenCredential(credentialsOrOptions)) {
    credential = credentialsOrOptions;
    credentialScopes = ["https://attest.azure.net/.default"];
    options  = clientOptions;
} else {
    options = credentialsOrOptions;
}

Reply:
perfect, updated as suggested